internal/commands/pull/command.go

```diff
@@ -168,30 +197,46 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 	}
 	ui.Print(terminal.NewTextLog("Saved app to disk"))
 
-	if cmd.inputs.IncludeDependencies {
+	if cmd.inputs.IncludeNodeModules || cmd.inputs.IncludePackageJSON || cmd.inputs.IncludeDependencies {
+		logStr := "as a node_modules archive"
+		if cmd.inputs.IncludePackageJSON {
+			logStr = "as a package.json file"
+		}
+
 		s := spinner.New(terminal.SpinnerCircles, 250*time.Millisecond)
-		s.Suffix = " Fetching dependencies archive..."
+		s.Suffix = fmt.Sprintf(" Fetching dependencies %s...", logStr)
+
+		var packageJSONMissing bool
 
 		exportDependencies := func() error {
 			s.Start()
 			defer s.Stop()
 
-			archiveName, archivePkg, err := clients.Realm.ExportDependencies(app.GroupID, app.ID)
+			fileName, file, err := cmd.exportDependencies(clients, app)
 			if err != nil {
 				return err
 			}
 
+			if cmd.inputs.IncludePackageJSON && fileName != local.NamePackageJSON {
+				packageJSONMissing = true
+			}
+
 			return local.WriteFile(
-				filepath.Join(pathBackend, local.NameFunctions, archiveName),
+				filepath.Join(pathBackend, local.NameFunctions, fileName),
 				0666,
-				archivePkg,
+				file,
 			)
 		}
 
 		if err := exportDependencies(); err != nil {
 			return err
 		}
-		ui.Print(terminal.NewTextLog("Fetched dependencies archive"))
+
+		if packageJSONMissing {
+			logStr = "as a node_modules archive"
+			ui.Print(terminal.NewWarningLog("The package.json file was not found, exporting a node_modules archive instead"))
```

Comment:
mostly, just try not to use the word "export" here. also, considering we will have this after "Fetched dependencies as a node_modules archive", we should avoid the use of present/progressive tenses

```suggestion
			ui.Print(terminal.NewWarningLog("The package.json file was not found, a node_modules archive was written instead"))
```